src/cargo/ops/cargo_output_metadata.rs

```diff
@@ -115,27 +115,46 @@ fn build_resolve_graph(
     let target_data = RustcTargetData::new(ws, &requested_kinds)?;
     // Resolve entire workspace.
     let specs = Packages::All.to_package_id_specs(ws)?;
+    let requested_features = RequestedFeatures::from_command_line(
+        &metadata_opts.features,
+        metadata_opts.all_features,
+        !metadata_opts.no_default_features,
+    );
```

Comment:
I sort of forget the interactions with this and the feature resolver we have, @ehuss I think will need to weigh in on this. 

From a code-style perspective, though, this is practically the same as `ResolveOpts` below so it might be nice to have one constructed from the other to avoid passing all these options twice.

Reply:
I just went ahead and fixed this up.